pkg/frr/frr.go

```diff
@@ -514,12 +510,17 @@ func tearDownVrf(vrf *infradb.Vrf) bool {
 		log.Printf("FRR Deleted event")
 		delCmd1 := fmt.Sprintf("no router bgp %+v vrf %s", localas, path.Base(vrf.Name))
 		delCmd2 := fmt.Sprintf("no vrf %s", path.Base(vrf.Name))
-		_, err = Frr.FrrBgpCmd(ctx, fmt.Sprintf("configure terminal\n %s\n exit\n", delCmd1))
-		if err != nil {
+		data, err = Frr.FrrBgpCmd(ctx, fmt.Sprintf("configure terminal\n %s\n exit\n", delCmd1))
+		if strings.Contains(data, "Can't find BGP instance") { // Trying to delete non exist VRF return true
+			return true
+		}
+		if err != nil || checkFrrResult(data, false) {
+			log.Printf("FRR: Error  %s\n", data)
 			return false
 		}
-		_, err = Frr.FrrZebraCmd(ctx, fmt.Sprintf("configure terminal\n %s\n exit\n", delCmd2))
-		if err != nil {
+		data, err = Frr.FrrZebraCmd(ctx, fmt.Sprintf("configure terminal\n %s\n exit\n", delCmd2))
+		if err != nil || checkFrrResult(data, false) {
+			log.Printf("FRR: Error  %s\n", data)
```

Comment:
Does data contain error description?

Reply:
Yes data contains the complete FRR command output.

Reply:
would it be better in case of an error, in FrrBgpCmd, just add data content into err and return empty first argument?

Reply:
Done